lib/puma.rb

```diff
@@ -39,6 +39,20 @@ def self.ssl?
     HAS_SSL
   end
 
+  def self.abstract_unix_socket?
```

Comment:
Should this be in `detect`?

Reply:
I could make it more clear, but I thought `detect` would be for OS dependent constants/methods, and Puma was used for constants/methods that could only be determined after something was required.

To check 'ssl', 'puma/puma_http11' needs to be required.  To check `UNIXSocket` (and abstract), we need to require 'socket'?